zhmcclient_mock/_hmc.py

```diff
@@ -0,0 +1,1006 @@
+                                         grandchild_list)
+
+    @staticmethod
+    def _assert_op_key(i, op, key):
+        if key not in op:
+            raise ValueError("Missing '{}' key in operations item #{}".
+                             format(key, i))
+
+
+class FakedActivationProfileManager(FakedBaseManager):
+    """
+    A manager for faked Activation Profile resources within a faked HMC (see
+    :class:`zhmcclient_mock.FakedHmc`).
+
+    Derived from :class:`zhmcclient_mock.FakedBaseManager`, see there for
+    common methods and attributes.
+    """
+
+    def __init__(self, hmc, cpc, profile_type):
+        activation_profiles = profile_type + '-activation-profiles'
+        super(FakedActivationProfileManager, self).__init__(
+            hmc=hmc,
+            parent=cpc,
+            resource_class=FakedActivationProfile,
+            base_uri=cpc.uri + '/' + activation_profiles,
+            oid_prop='object-id',
+            uri_prop='object-uri')
+        self._profile_type = profile_type
+
+    @property
+    def profile_type(self):
+        """
+        Type of the activation profile ('reset', 'image', 'load').
+        """
+        return self._profile_type
+
+
+class FakedActivationProfile(FakedBaseResource):
+    """
+    A faked Activation Profile resource within a faked HMC (see
+    :class:`zhmcclient_mock.FakedHmc`).
+
+    Derived from :class:`zhmcclient_mock.FakedBaseResource`, see there for
+    common methods and attributes.
+    """
+
+    def __init__(self, manager, properties):
+        super(FakedActivationProfile, self).__init__(
+            manager=manager,
+            properties=properties)
+
+
+class FakedAdapterManager(FakedBaseManager):
+    """
+    A manager for faked Adapter resources within a faked HMC (see
+    :class:`zhmcclient_mock.FakedHmc`).
+
+    Derived from :class:`zhmcclient_mock.FakedBaseManager`, see there for
+    common methods and attributes.
+    """
+
+    def __init__(self, hmc, cpc):
+        super(FakedAdapterManager, self).__init__(
+            hmc=hmc,
+            parent=cpc,
+            resource_class=FakedAdapter,
+            base_uri=self.api_root + '/adapters',
+            oid_prop='object-id',
+            uri_prop='object-uri')
+
+
+class FakedAdapter(FakedBaseResource):
+    """
+    A faked Adapter resource within a faked HMC (see
+    :class:`zhmcclient_mock.FakedHmc`).
+
+    Derived from :class:`zhmcclient_mock.FakedBaseResource`, see there for
+    common methods and attributes.
+    """
+
+    def __init__(self, manager, properties):
+        super(FakedAdapter, self).__init__(
+            manager=manager,
+            properties=properties)
+        self._ports = FakedPortManager(hmc=manager.hmc, adapter=self)
+
+    @property
+    def ports(self):
+        """
+        The Port resources of this Adapter
+        (:class:`~zhmcclient_mock.FakedPort`).
+        """
+        return self._ports
+
+
+class FakedCpcManager(FakedBaseManager):
```

Comment:
Again, start() and stop() at the external APi will work. They become post() calls on the (internal) UriHandler. the handler for those is not yet implemented at this point, but when it is, it will simply change the status of the partition object in the FakedHmc object via direct properties update.